opensearch_py_ml/sentence_transformer_model/sentencetransformermodel.py

```diff
@@ -743,6 +781,100 @@ def set_up_accelerate_config(
         with open(file_path, "w") as file:
             yaml.dump(default_file, file)
 
+    def make_model_config_json(
+        self,
+        model_name: str = None,
+        version_number: int = 1,
+        embedding_dimension: int = 768,
+        all_config: str = None,
+        model_type: str = None,
+    ) -> None:
+        """
+        parse from config.json file of pre-trained hugging-face model to generate a model_config.json file. If all required
+        fields are given by users, use the given parameters and will skip reading the config.json,
+
+        Parameters
+        ----------
+        model_name: str = None
+            Optional, The name of the model. If None, default to parse from model id, for example,
+            'msmarco-distilbert-base-tas-b'
+        version_number: int = 1,
+            Optional, The version number of the model. If None, default to be 1.
+        embedding_dimension: int = 768,
+            Optional, the embedding_dimension of the model. If None, parse embedding_dimension from the config file of
+             pre-trained hugging-face model, if not found, default to be 768.
+        all_config: dict = None,
+            Optional, the all_config of the model. If None, parse all contents from the config file of pre-trained
+            hugging-face model.
+        model_type: str = None,
+            Optional, the model_type of the model. If None, parse model_type from the config file of pre-trained
+            hugging-face model.
+
+        Returns
+        -------
+        None
+        """
+        folder_path = self.folder_path
+        config_json_file_path = os.path.join(folder_path, "config.json")
+        if model_name is None:
+            model_name = self.model_id.split("/")[-1]
```

Comment:
ml-commons allow "/" in model name, for example "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b" is a valid model name. Not a bug/issue at all, just feel keep the original huggingface model id could be easier for user later. For example what if there will be another huggingface model "xyz/msmarco-distilbert-base-tas-b" published by "xyz" company? Two models with same name "msmarco-distilbert-base-tas-b" is hard to track.

Reply:
I tried keeping the "/" but using ml_common.upload gave me an error that the model expects one fields but got two, I guess the slack will make the url messed up. check [here](https://github.com/opensearch-project/opensearch-py-ml/blob/f353dfaa038f1c153b1ca1966477dcbb72eb98dd/opensearch_py_ml/ml_commons_integration/upload/ml_common_model_uploader.py#L109): 
                        url=f"{ML_BASE_URI}/models/{model_id}/chunk/{i}",

Reply:
> url=f"{ML_BASE_URI}/models/{model_id}/chunk/{i}",

`model_id` is uuid, not the model name, for example `POST /_plugins/_ml/models/uTeUyoMBCSheqCFfL8Bg/chunk/0`

Reply:
in line 820. changed to line 820 works fine in testing. Changed to `model_name = self.model_id`.